src/main/js/Upload/FileUpload.tsx

```diff
@@ -217,15 +212,24 @@ class FileUpload extends React.Component<Props, State> {
             </Header>
           )}
           <Breadcrumb repository={repository} baseUrl={baseUrl} path={path} revision={revision} />
-          <FileMetaData path={path} changePath={this.changePath} />
+          <FileMetaData
+            path={path}
+            changePath={changedPath => {
+              this.setState({
+                path: changedPath,
+                shouldValidate: false
+              });
+            }}
+            onBlur={() => this.setState({ shouldValidate: true })}
+          />
           <FileUploadDropzone fileHandler={this.handleFile} disabled={loading} />
         </Border>
         {files && files.length > 0 && (
           <FileUploadTable files={files} removeFileEntry={this.removeFileEntry} disabled={loading} />
         )}
         <ExtensionPoint
           name="editorPlugin.file.upload.validation"
-          props={{ repository, files, path, validateFiles: (disable: boolean) => this.setState({ valid: disable }) }}
+          props={{ repository, files, path, shouldValidate }}
```

Comment:
The property `shouldValidate` is named for the current use case in the file lock plugin, but might be used otherwise by other extensions. What about naming it for what it does: it indicates, that the path is being edited right now and therefore might change again. So maybe `isPathEditing` like `isLoading` in the query api?
(of course this would mean that the values have to be inverted)